.github/workflows/mirror-dev.yml

```diff
@@ -0,0 +1,34 @@
+name: Mirror dev to GitLab
+
+on:
+  push:
+    branches:
+      - dev
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - dev
+
+jobs:
+  mirror:
+    if: github.repository == 'accessibility-exchange/platform'
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Git
+        run: |
+          git config --global user.name "${{ vars.GIT_USER }}"
+          git config --global user.email "${{ vars.GIT_EMAIL }}"
+          git remote add gitlab ${{ vars.GITLAB_URL }}
+
+      - name: Push to GitLab
+        run: |
+          mkdir -p ~/.ssh
+          echo "${{ secrets.GITLAB_SSH_KEY }}" > ~/.ssh/id_rsa
+          chmod 600 ~/.ssh/id_rsa
+          ssh-keyscan gitlab.com >> ~/.ssh/known_hosts
```

Comment:
Instead of setting up the ssh key like this, I wonder if an approach as documented in this blog post ( https://www.webfactory.de/blog/use-ssh-key-for-private-repositories-in-github-actions ) would be better. Supposedly it doesn't write the private key to disk which will add a bit more security. They have also created a GitHub action you can use, if you'd like to investigate that.

Reply:
Thanks this was a great reference. I will be changing to use their action.